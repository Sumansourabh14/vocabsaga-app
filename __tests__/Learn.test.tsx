import Learn from "@/app/(tabs)/learn";
import { render } from "@testing-library/react-native";

describe("Learn", () => {
  it("renders without crashing", () => {
    const { getByText } = render(<Learn />);

    getByText("Learn words");
  });

  it("learn component snapshot", () => {
    const snapshot = render(<Learn />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
