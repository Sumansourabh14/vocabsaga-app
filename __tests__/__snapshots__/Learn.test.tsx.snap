// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Learn learn component snapshot 1`] = `
<RNCSafeAreaView
  className="flex-1"
  edges={
    {
      "bottom": "additive",
      "left": "additive",
      "right": "additive",
      "top": "additive",
    }
  }
>
  <Text
    className="text-4xl font-playfairBold text-center py-4"
    style={
      {
        "color": "#000",
      }
    }
  >
    Learn words
  </Text>
  <View
    className="mt-6 px-6"
  >
    <View
      accessibilityLabel="Learn new words through passages"
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      className="rounded-xl border-hairline px-6 py-8 mb-4 flex-row items-center"
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#000",
        }
      }
    >
      <Text
        allowFontScaling={false}
        className="mr-6"
        selectable={false}
        style={
          [
            {
              "color": "#4B5563",
              "fontSize": 48,
            },
            undefined,
            {
              "fontFamily": "ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        îª§
      </Text>
      <View
        className="flex-1"
      >
        <Text
          className="text-lg font-interBold mb-2"
          style={
            {
              "color": "#000",
            }
          }
        >
          Learn through passages
        </Text>
        <Text
          className="font-inter"
          style={
            {
              "color": "#595959",
            }
          }
        >
          Understand new words in context by reading sentences
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
